### PR TITLE
(chore) Refactor DayPickerProps types, remove uncontrolled selections

### PR DESCRIPTION
**Pull request by gpbl**
We have been dealing with the "controlled" or "uncontrolled" behavior but I don't think it is necessary in case of selections –  by looking to our examples, we always need a "controlled" mode. This however was a bit confusing with the current typings of the props.

* Renamed `DayPickerBase` to `DayPickerProps` and made the four interfaces for the selection modes (`DayPickerSingleProps`,`DayPickerMultipleProps`, etc.) extend `DayPickerProps` itself.
* Removed the `defaultSelected` prop.

### CODE REVIEW

**Review by tjfred35 — COMMENTED** [review]
I like the direction this is going in 👍

File: packages/react-day-picker/src/types/DayPickerProps.ts [hunk 1]
```diff
@@ -1,15 +1,240 @@
...
+  /**
+   * A map of labels creators used for the ARIA labels attributes.
+   */
+  labels?: Partial<Labels>;
+
+  /**
+   * The text direction of the calendar. Use `ltr` for left-to-right (default)
+   * or `rtl` for right-to-left.
+   */
+  dir?: string;
```
Comment (tjfred35) [inline packages/react-day-picker/src/types/DayPickerProps.ts:215]: Could this be:
```
dir?: 'ltr' | 'rtl'
```

Comment (gpbl) [reply packages/react-day-picker/src/types/DayPickerProps.ts:215]: I'd keep the HTML default for the HTMLDivElement which is `string`.

Comment (tjfred35) [reply packages/react-day-picker/src/types/DayPickerProps.ts:215]: Ok fair

**Comment by tjfred35** [comment]
I can’t think of a reason to use defaultSelected, but if I can we can always add it back. I do have a reason to use defaultMonth.

**Comment by gpbl** [comment]
> I can’t think of a reason to use defaultSelected, but if I can we can always add it back. I do have a reason to use defaultMonth.

Sure, @tjfred35 `defaultMonth` is still there.

**Comment by tjfred35** [comment]
Is “being able to play with DayPicker out of the box without wiring up an external React.useState” a reason to continue to support an uncontrolled mode?

File: website/examples/navigation-disabled.tsx [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+import { DayPicker } from "react-day-picker";
 
 export function Example() {
   return (
-    <DayPicker fromYear={2015} toYear={2025} disableNavigation mode="single" />
+    <DayPicker
+      fromYear={2015}
+      toYear={2025}
+      disableNavigation
+      mode="single"
```
Comment (tjfred35) [inline website/examples/navigation-disabled.tsx:10]: Here is an uncontrolled example that may no longer work as intended

File: website/examples/customization-fixedweeks.tsx [hunk 3]
```diff
@@ -2,5 +2,5 @@ import React from "react";
 import { DayPicker } from "react-day-picker";
 
 export function Example() {
-  return <DayPicker showOutsideDays fixedWeeks />;
+  return <DayPicker fixedWeeks />;
```
Comment (tjfred35) [inline website/examples/customization-fixedweeks.tsx:5]: I thought fixedWeeks also required showOutsideDays. Did you change that?

File: website/docs/guides/custom-selections.md [hunk 4]
```diff
@@ -1,17 +1,14 @@
...
-to custom selection mode or – to keep the built-in modes – use a custom Day component.
+To implement a selection behavior by yourself, you can either switch to custom
+selection mode or – to keep the built-in modes – use a custom Day component.
 
 ## Switching to the custom mode
 
-1. set the selection mode to `mode="custom"`
-2. store the selected dates in the parent’s state
-3. use the `selected` and `onDayClick` props
+To switch to custom selection, the `mode` propo to `custom`:
```
Comment (tjfred35) [inline website/docs/guides/custom-selections.md:11]: “set the” … “prop”…

File: website/examples/custom-selection-single.tsx [hunk 5]
```diff
@@ -1,19 +1,18 @@
...
-    : "Please pick a day.";
+  let footer = "Please pick a day.";
+  if (selected) {
+    footer = `You picked ${selected.toLocaleDateString()}.`;
+  }
   return (
     <DayPicker
-      mode="custom"
-      selected={selectedDay}
-      onDayClick={handleDayClick}
```
Comment (tjfred35) [inline website/examples/custom-selection-single.tsx:16]: We don’t need onDayClick anymore? I’d like to test this a little bit. I’m not looking at the code right now, but I’m unfamiliar with the code calls onSelect in the custom mode

File: website/examples/customization-fixedweeks.tsx (hunk 3 above)
Comment (gpbl) [reply website/examples/customization-fixedweeks.tsx]: Good catch thanks, restored in b2ba94237a8427021dac77a4baaafd953352ddef.

File: website/docs/guides/custom-selections.md (hunk 4 above)
Comment (gpbl) [reply website/docs/guides/custom-selections.md]: Thanks fixed in b508d948432d138abed260923935b652f7a5fca4.

File: website/examples/custom-selection-single.tsx (hunk 5 above)
Comment (gpbl) [reply website/examples/custom-selection-single.tsx:16]: Sorry for the confusion, that was a wrong copy/paste. The example is restored in dd748c5851f53e52d20326c7cb4d9c874c934cf1.

File: website/examples/navigation-disabled.tsx (hunk 2 above)
Comment (gpbl) [reply website/examples/navigation-disabled.tsx:10]: I prefer this version with no "magic": without specifying `selected` or `onSelect`, nothing happens. 

(I will remove the mode prop from this example)

**Comment by gpbl** [comment]
> Is “being able to play with DayPicker out of the box without wiring up an external React.useState” a reason to continue to support an uncontrolled mode?

To me yes so far – what do you think? :)

**Comment by gpbl** [comment]
@tjfred35 I could fix some bugs that should address your concerns about the click events (without the tests this is getting complicated).

**Comment by gpbl** [comment]
I'm merging this as it is required for the next PRs.

**Pull request merged**